hkmc2/shared/src/main/scala/hkmc2/codegen/llir/Llir.scala

```diff
@@ -174,33 +102,100 @@ enum Node:
   case LetExpr(name: Local, expr: Expr, body: Node)
   case LetMethodCall(names: Ls[Local], cls: Local, method: Local, args: Ls[TrivialExpr], body: Node)
   case LetCall(names: Ls[Local], func: Local, args: Ls[TrivialExpr], body: Node)
-
-  override def toString: String = show
-
-  def show: String = toDocument.toString
-
-  def toDocument: Document =
+  def show = LlirDebugPrinter.mkDocument(this).toString
+
+trait LlirPrinting:
```

Comment:
Traits are more expensive to compile than classes and there is no reason to make this one a trait.
```suggestion
class LlirPrinting:
```